src/lib/service/git-service.js

```diff
@@ -0,0 +1,24 @@
+const git = require("../git");
+const { logger } = require("../common");
+
+/**
+ * It treats a git hash from a remote repositry.
+ *
+ * @param {string} repositoryUrl the repository URL
+ * @param {string} branch the branch to get the hash from
+ * @param {number} hashLength the length of hash to return back
+ */
+async function getRemoteSha(repositoryUrl, branch, hashLength = 7) {
+  try {
+    const remoteHashValue = await git.remoteSha(repositoryUrl, branch);
```

Comment:
I'm not familiar with this code so I don't know if the question makes sense but is this invoking normal git CLI command or is it a GitHub remote API call?
I remember in the past we had failures related to the number of API call so I want to double check if this is increasing the number of calls we do or not.

Reply:
it is actually a very good question. This is executing the `remoteSha` method from `git.js` file which basically at the end executes the git CLI tool from the filesystem.